src/esp/nav/PathFinder.cpp

```diff
@@ -946,9 +953,46 @@ vec3f PathFinder::Impl::getRandomNavigablePoint(const int maxTries /*= 10*/) {
     ESP_ERROR() << "Failed to getRandomNavigablePoint.  Try increasing max "
                    "tries if the navmesh is fine but just hard to sample from";
     return vec3f::Constant(Mn::Constants::nan());
-  } else {
-    return pt;
   }
+  return pt;
+}
+
+vec3f PathFinder::Impl::getRandomNavigablePointAroundSphere(
+    const vec3f& circleCenter,
+    const float radius,
+    const int maxTries) {
+  if (getNavigableArea() <= 0.0)
+    throw std::runtime_error(
+        "NavMesh has no navigable area, this indicates an issue with the "
+        "NavMesh");
+
+  vec3f pt = vec3f::Constant(Mn::Constants::nan());
+  dtPolyRef start_ref = 0;  // ID to start our search
+  dtStatus status = navQuery_->findNearestPoly(
+      circleCenter.data(), vec3f{radius, radius, radius}.data(), filter_.get(),
+      &start_ref, pt.data());
+  if (!dtStatusSucceed(status) || std::isnan(pt[0]) ||
+      (pt - circleCenter).norm() > radius) {
+    ESP_ERROR()
+        << "Failed to getRandomNavigablePoint. No polygon found within radius";
+    return vec3f::Constant(Mn::Constants::nan());
+  }
+  int i = 0;
+  for (; i < maxTries; ++i) {
+    dtPolyRef rand_ref = 0;
+    status = navQuery_->findRandomPointAroundCircle(
+        start_ref, circleCenter.data(), radius, filter_.get(), frand, &rand_ref,
+        pt.data());
+    if (dtStatusSucceed(status) && (pt - circleCenter).norm() <= radius) {
+      break;
+    }
+  }
+  if (i == maxTries) {
+    ESP_ERROR() << "Failed to getRandomNavigablePoint.  Try increasing max "
+                   "tries if the navmesh is fine but just hard to sample from";
+    return vec3f::Constant(Mn::Constants::nan());
```

Comment:
Also wasn't sure if I should just return the nearest point here if the "random" selection of neighboring points failed for some reason. I figured to return a NaN just to be consistent even if it  does find a valid point (AKA the nearest point only).